tests/test_cellpylib.py

```diff
@@ -450,6 +481,21 @@ def test_binary_rule_powers_of_two_default(self):
 
         np.testing.assert_equal(expected.tolist(), actual.tolist())
 
+    def test_plot(self):
+        # this test ensures that the following code can run successfully without issue
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", UserWarning)
```

Comment:
These seem a bit repetitive and just suppressing a warning might not be what you want to do here. How about the following?

```python
PLOT_WARNING_MATCH = "the warning message to match"

with pytest.warns(UserWarning, match=PLOT_WARNING_MATCH):
    cpl.plot(...)
```

Instead of suppressing the error message, this tests for it being *present*, indirectly testing that `plt.show` is invoked (which, I assume, causes a warning to be emitted when using non-interactive backends like here in the test). In case the warning will no longer be raised, the test will fail, making you aware that something changed.